### PR TITLE
std.file docs: Tweak dirEntries examples

### PR DESCRIPTION
**Pull request by ntrel**
Use selective imports for `listdir` example, not fully qualified calls.
Don't use `args`, use current directory for working runnable example.

Move parallel dmd example into a documented unittest. Use `pattern` argument instead of filter & endsWith for simplicity. Note the `listdir` example already uses `map`.

Only run these 'tests' for `version(StdDdoc)`.

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -4966,28 +4953,49 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
-        import std.algorithm;
-        import std.array;
-        import std.file;
-        import std.path;
+        import std.algorithm.iteration : filter, map;
+        import std.array : array;
+        import std.path : baseName;
 
-        return std.file.dirEntries(pathname, SpanMode.shallow)
+        return dirEntries(pathname, SpanMode.shallow)
```
Comment (PetarKirov) [inline std/file.d:4965]:
```suggestion
        return pathname.dirEntries(SpanMode.shallow)
```

File: std/file.d [hunk 2]
```diff
@@ -4966,28 +4953,49 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
             .filter!(a => a.isFile)
-            .map!(a => std.path.baseName(a.name))
+            .map!(a => baseName(a.name))
             .array;
     }
 
-    void main(string[] args)
-    {
+    void main() {
```
Comment (PetarKirov) [inline std/file.d:4971]: The style check fails because each curly brace must be on its own line.

File: std/file.d [hunk 3]
```diff
@@ -4966,28 +4953,49 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
         import std.stdio;
 
-        string[] files = listdir(args[1]);
-        writefln("%s", files);
-     }
+        string[] files = listdir("");
```
Comment (PetarKirov) [inline std/file.d:4974]: 1. Could use UFCS
2. Using `"."` makes it more clear that we're listing the contents of the current directory

```suggestion
        string[] files = ".".listdir();
```

File: std/file.d [hunk 4]
```diff
@@ -4966,28 +4953,49 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
+    import std.stdio;
+    import std.parallelism;
+    import std.process;
+
+    // Iterate over all *.d files in current directory and all its subdirectories
+    auto dFiles = dirEntries("", "*.d", SpanMode.depth);
+    foreach (d; dFiles)
+        writeln(d.name);
+
+    foreach (d; parallel(dFiles, 1)) // process each entry in a new thread
```
Comment (PetarKirov) [inline std/file.d:4992]:
```suggestion
    foreach (d; parallel(dFiles, 1)) // process each entry in a separate thread
```

`std.parallelism.parallel` uses a shared thread pool for the whole program, so there are at most `N+1` (IIRC) parallel invocations of `dmd`, where `N` is the number of CPU threads (e.g. 8, on a 4 core CPU with 2-way SMT). As was written before, the comment implied that there would be up to `dFiles.length` dmd processes in flight.

File: std/file.d [hunk 5]
```diff
@@ -4966,28 +4953,49 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
+    {
+        // compile source file
+        string cmd = "dmd -c "  ~ d.name;
```
Comment (PetarKirov) [inline std/file.d:4995]:
```suggestion
        string cmd = "dmd -c " ~ d.name;
```

**Comment by CyberShadow** [comment]
I would also like to suggest using UFCS in examples where appropriate, as it is closer to idiomatic D.

File: std/file.d [hunk 6]
```diff
@@ -4970,24 +4957,45 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
+    import std.stdio;
+    import std.parallelism;
+    import std.process;
+
+    // Iterate over all *.d files in current directory and all its subdirectories
+    auto dFiles = dirEntries(".", "*.d", SpanMode.depth);
+    foreach (d; dFiles)
+        writeln(d.name);
+
+    foreach (d; parallel(dFiles, 1)) // process each entry in a separate thread
```
Comment (CyberShadow) [inline std/file.d:4992]:
```suggestion
    foreach (d; dFiles.parallel(1)) // process each entry in a separate thread
```

**Comment by RazvanN7** [comment]
@ntrel I agree with the other reviewers, UFCS makes the code much nicer to read.

**Review by RazvanN7 — APPROVED** [review]
I'm gonna merge this as is and we can apply the required changes in a follow-up PR

File: std/file.d [hunk 7]
```diff
@@ -4970,24 +4957,45 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
-     }
+        string[] files = listdir(".");
+        writeln(files);
+    }
+}
+
+/// Compile D files in parallel:
+version (StdDdoc)
+@system unittest
+{
```
Comment (berni44) [inline std/file.d:4982]: I think, this unittest introduced [issue 21796](https://issues.dlang.org/show_bug.cgi?id=21796).

See [logfile of circle-ci](https://app.circleci.com/pipelines/github/dlang/phobos/3976/workflows/90ccdf2d-ff37-4e70-b7da-ca80d626c604/jobs/13524).

File: std/file.d [hunk 8]
```diff
@@ -4970,24 +4957,45 @@ auto dirEntries(string path, SpanMode mode, bool followSymlink = true)
...
+    auto dFiles = dirEntries(".", "*.d", SpanMode.depth);
+    foreach (d; dFiles)
+        writeln(d.name);
+
+    foreach (d; parallel(dFiles, 1)) // process each entry in a separate thread
+    {
+        // compile source file
+        string cmd = "dmd -c " ~ d.name;
+        writeln(cmd);
+        std.process.executeShell(cmd);
```
Comment (ibuclaw) [inline std/file.d:4997]: WHAT!

**Comment by ibuclaw** [comment]
Hi @ntrel - This PR caused a [regression](https://issues.dlang.org/show_bug.cgi?id=21796), and has been [reverted](https://github.com/dlang/phobos/pull/8009).  Please feel free to resend a version of this PR without the regressing part.  Thanks!

**Pull request merged**